ci/src/cI_config.ml

```diff
@@ -1,26 +1,31 @@
+open Datakit_github
 open! Astring
 
 type test = string CI_term.t
 
 type project = {
-  dashboards : CI_target.ID_Set.t;
-  tests : CI_target.Full.t -> test String.Map.t;
+  dashboards : CI_target.Set.t;
+  tests : CI_target.t -> test String.Map.t;
 }
 
 type t = {
   web_config : CI_web_templates.t;
-  projects : project CI_projectID.Map.t;
+  projects : project Repo.Map.t;
 }
 
-let id_of_branch name =
-  `Ref (Datakit_path.of_string_exn ("heads/" ^ name))
+let id_of_branch repo name = `Ref (repo, "heads" :: String.cuts ~sep:"/" name)
 
 let project ~id ?(dashboards=["master"]) tests =
-  let id = CI_projectID.of_string_exn id in
+  let id = match Repo.of_string id with
+    | None   -> failwith ("project: " ^ id)
```

Comment:
I don't think these errors are very clear. Might be worth giving `Repo.of_string` a `result` type. I can't think of any reason why you'd care that a repo name was invalid and not want to log the reason somewhere.

Failing that, `Utils.failf "Invalid repo ID %S" id` would be clearer.